crowbar_framework/app/models/service_object.rb

```diff
@@ -1631,9 +1631,29 @@ def equal_or_match_version(value, maybe_regexp)
       end
     end
 
-    cmp = lambda do |op, version|
-      (to_version.call(value).zip to_version.call(version)).all? do |v1, v2|
-        v1.send(op, v2)
+    cmp = lambda do |version1, version2|
+      version1.zip version2.each do |v|
+        ans = v[0] <=> v[1]
+        return ans if ans != 0
+      end
+      0
+    end
+
+    oper = lambda do |op, version|
```

Comment:
Again, just for readability: I'd make ` value` an argument of this, so we don't have to scroll back to see that it's an argument to the parent method.